### PR TITLE
add fast check for no leaks

### PR DESCRIPTION
**Pull request by Maxime2**
If we have no leaks we can finish testing straight away before setting interval timer and its first tick arrives.

### CODE REVIEW

**Comment by fortytw2** [comment]
Awesome! Thanks for your contribution :+1: 

Can you extract the duplicated section into another function? L133-141 are the exact same as L148-156 

I'll merge and cut a release once you do

File: leaktest.go [hunk 1]
```diff
@@ -94,6 +94,21 @@ func interestingGoroutines(t ErrorReporter) []*goroutine {
 	return gs
 }
 
+// leakedGoroutines returns all goroutines we are considereing leaked and
+// the boolean flag indicating if no leaks detected
+func leakedGoroutines(orig map[uint64]bool, interesting []*goroutine) ([]string, bool) {
+	var leaked []string
+	leaked = make([]string, 0)
```
Comment (fortytw2) [inline leaktest.go]: you don't need to do this line!

Comment (Maxime2) [reply leaktest.go]: corrected!

Comment (Maxime2) [reply leaktest.go]: I have squashed commits with minor corrections into one introducing leakedGoroutines.

**Comment by fortytw2** [comment]
Thanks @Maxime2!

**Pull request merged**